runtime/fundamentals/testing.md

```diff
@@ -0,0 +1,757 @@
+deno test util/
+
+# Run just my_test.ts
+deno test my_test.ts
+
+# Run test modules in parallel
+deno test --parallel
+
+# Pass additional arguments to the test file
+deno test my_test.ts -- -e --foo --bar
+```
+
+## Test Steps
+
+Deno also supports test steps, which allow you to break down tests into smaller,
+manageable parts. This is useful for setup and teardown operations within a
+test:
+
+```ts
+Deno.test("database operations", async (t) => {
+  await t.step("insert user", async () => {
+    // Insert user logic
+  });
+  await t.step("insert book", async () => {
+    // Insert book logic
+  });
+});
+```
+
+## Command line filtering
+
+Deno allows you to run specific tests or groups of tests using the `--filter`
+option on the command line. This option accepts either a string or a pattern to
+match test names.
+
+Consider the following tests:
+
+```ts
+Deno.test({ name: "my-test", fn: myTest });
+Deno.test({ name: "test-1", fn: test1 });
+Deno.test({ name: "test-2", fn: test2 });
+```
+
+### Filtering by string
+
+To run all tests that contain the word "test" in their names, use:
+
+```sh
+deno test --filter "test" tests/
+```
+
+This command will execute `my-test`, `test-1`, and `test-2` because they all
+include "test" in their names.
+
+### Filtering by Pattern
+
+To run tests that match a specific pattern, use:
+
+```sh
+deno test --filter "/test-*\d/" tests/
+```
+
+This command will run `test-1` and `test-2` because they match the pattern
+`test-*` followed by a digit.
+
+To indicate that you are using a pattern (regular expression), wrap your filter
+value with forward slashes `/`, much like JavaScript’s syntax for regular
+expressions.
+
+### Including and excluding paths in the configuration file
+
+You can also filter tests by specifying paths to include or exclude in the Deno
+configuration file.
+
+For example, if you want to only test `src/fetch_test.ts` and
+`src/signal_test.ts` and exclude everything in `out/`:
+
+```json
+{
+  "test": {
+    "include": [
+      "src/fetch_test.ts",
+      "src/signal_test.ts"
+    ]
+  }
+}
+```
+
+Or more likely:
+
+```json
+{
+  "test": {
+    "exclude": ["out/"]
+  }
+}
+```
+
+Then running `deno test` in the same directory tree as the configuration file
+will take these options into account.
+
+## Test definition filtering
+
+Deno provides two options for filtering tests within the test definitions
+themselves: ignoring tests and focusing on specific tests.
+
+### Filtering Out (Ignoring Tests)
+
+You can ignore certain tests based on specific conditions using the `ignore`
+boolean in the test definition. If `ignore` is set to `true`, the test will be
+skipped. This is useful, for example, if you only want a test to run on a
+specific operating system.
+
+```ts
+Deno.test({
+  name: "do macOS feature",
+  ignore: Deno.build.os !== "darwin", // This test will be ignored if not running on macOS
+  fn() {
+    // do MacOS feature here
+  },
+});
+```
+
```

Comment:
Add an example for the shorthand syntax, for when you don't want to pass arguments and just skip tests for any reason:

```ts
Deno.test.ignore("my test", () => {
  // some test code
})
```